.github/workflows/main.yml

```diff
@@ -0,0 +1,120 @@
+name: CI
+
+on:
+  push:
+    branches: [main, dev, test]
+
+jobs:
+
+  # Set the name of the environment based on the branch. 
```

Comment:
(Since you figured out combining the actions into one file, we can update the other repositories with Actions to match)